app/models/pipeline_run.rb

```diff
@@ -345,6 +346,11 @@ def create_run_stages
       job_command_func: 'experimental_command'
     )
 
+    dag_names = [DAG_NAME_HOST_FILTER, DAG_NAME_ALIGNMENT, DAG_NAME_POST_PROCESS, DAG_NAME_EXPERIMENTAL]
+    run_stages.each_with_index do |stage, i|
+      stage.dag_json = generate_dag_json(self, dag_names[i])
```

Comment:
Positional loops iterating two arrays simultaneously can be dangerous if someone change the order in any of these arrays (like inserting an item in only one of them).

You could use a dictionary, but I believe it is easier to actually invoke them directly in the commands above, like:

```
    run_stages << PipelineRunStage.new(
      step_number: 1,
      name: PipelineRunStage::HOST_FILTERING_STAGE_NAME,
      job_command_func: 'host_filtering_command',
      dag_json: generate_dag_json(self, DagJsonHelper::DAG_NAME_HOST_FILTER)
    )
```

Reply:
Actually, if you make the change this way, you don't actually need this `generate_dag_json` method. You could directly invoke the methods with the correct names (like `DagJsonHelper#host_filtering_dag_attributes`)